packages/binding-coap/src/coaps-client.ts

```diff
@@ -105,11 +105,11 @@ export default class CoapsClient implements ProtocolClient {
         complete?: () => void
     ): Promise<Subscription> {
         return new Promise<Subscription>((resolve, reject) => {
-            let requestUri = url.parse(form.href.replace(/$coaps/, "https"));
+            const requestUri = url.parse(form.href.replace(/$coaps/, "https"));
```

Comment:
Can't we solve this? it should be siply `url.parse` => `new Url`. Am I missing something?

Reply:
The URL object doesn't have `query` property. Sometimes we are using it and sometimes not. To be on the safe side, I've decided to leave it as it is (at least for now).

Reply:
I ran into a similar problem in `node-coap` where I added a couple of dedicated function for retrieving the fields from the `URL` object: https://github.com/mcollina/node-coap/commit/71a90787b4960ebd36f62401338e81b057a0c58f Instead of `query` you can use the `search` property by removing the leading `?`.

This is obviously not very elegant, though. Maybe using something like [uri-js](https://www.npmjs.com/package/urijs) could be a better solution instead.

Reply:
@JKRhb Thanks for your comment!
@relu91 @danielpeintner Which way to go?

Reply:
Mhh, no strong opinion either way.

@fatadel I let you to decide to work on this in this PR or create an issue to work on it in a follow-up PR.
@relu91 any preference?

Reply:
The point of this PR is improving es-lint errors. So I would go with the constructor and fix it here. If we are using the query parameter @JKRhb propose a good algorithm. However, we might keep track of IPv6 issues.

Reply:
Done ✅